hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/RolloutListGrid.java

```diff
@@ -301,17 +340,29 @@ protected void setColumnHeaderNames() {
         if (permissionChecker.hasRolloutUpdatePermission()) {
             getColumn(UPDATE_OPTION).setHeaderCaption(i18n.get("header.action.update"));
         }
+
         if (permissionChecker.hasRolloutCreatePermission()) {
             getColumn(COPY_OPTION).setHeaderCaption(i18n.get("header.action.copy"));
         }
 
+        getColumn(DELETE_OPTION).setHeaderCaption(i18n.get("header.action.delete"));
+
+        final HeaderCell join = joinColumns();
+        join.setText(i18n.get("header.action"));
+    }
+
+    private HeaderCell joinColumns() {
         HeaderCell join;
-        if (permissionChecker.hasRolloutUpdatePermission()) {
-            join = getDefaultHeaderRow().join(RUN_OPTION, PAUSE_OPTION, UPDATE_OPTION, COPY_OPTION);
+        if (permissionChecker.hasRolloutUpdatePermission() && permissionChecker.hasRolloutCreatePermission()) {
+            join = getDefaultHeaderRow().join(RUN_OPTION, PAUSE_OPTION, UPDATE_OPTION, COPY_OPTION, DELETE_OPTION);
+        } else if (permissionChecker.hasRolloutUpdatePermission()) {
+            join = getDefaultHeaderRow().join(RUN_OPTION, PAUSE_OPTION, UPDATE_OPTION, DELETE_OPTION);
+        } else if (permissionChecker.hasRolloutCreatePermission()) {
+            join = getDefaultHeaderRow().join(RUN_OPTION, PAUSE_OPTION, COPY_OPTION, DELETE_OPTION);
         } else {
             join = getDefaultHeaderRow().join(RUN_OPTION, PAUSE_OPTION);
         }
-        join.setText(i18n.get("header.action"));
```

Comment:
If you remove this line, then you could return directly and need no variable `HeaderCell join;`. This is unnecessary and then you need just `if`s and no `else if`s 👍